src/Pact/Runtime/Capabilities.hs

```diff
@@ -153,69 +168,38 @@ acquireCapability i af scope cap capMF test = go
 
 
 checkManaged
-  :: ApplyMgrFun e
-  -> Capability
-  -> Maybe (Def Ref)
-  -> Eval e (Maybe (Either String (CapSlot Capability)))
-checkManaged _ _ Nothing = return Nothing
-checkManaged applyF cap (Just mgrFun) = use (evalCapabilities . capManaged) >>= go
+  :: HasInfo i
+  => i
+  -> ApplyMgrFun e
+  -> UserCapability
+  -> Def Ref
+  -> Eval e (Maybe [UserCapability])
+checkManaged i applyF cap@SigCapability{..} cdef
+  | isManaged cdef = use (evalCapabilities . capManaged) >>= go . S.toList
+  | otherwise = return Nothing
   where
+    go [] = evalError' i $ "Managed capability not installed: " <> pretty cap
+    go (mc@ManagedCapability{..}:mcs) = case decomposeManaged' _mcManageParamIndex cap of
```

Comment:
same here

Reply:
this is monadic on the results of the test as there is more than one failure mode, so not going to address this in interests of time